fetch.bs

```diff
@@ -2690,16 +2693,104 @@ functionality.
 <dfn export for="fetch record" id=concept-fetch-record-fetch>controller</dfn> (a
 <a for=/>fetch controller</a> or null).
 
+<p>A <dfn export>deferred fetch record</dfn> is a <a for=/>struct</a> used to maintain state needed
+to invoke a fetch at a later time, e.g., when a <code>Document</code> is unloaded or backgrounded. It
+has the following <a for=struct>items</a>:
+
+<dl>
+ <dt><dfn export for="deferred fetch record">request</dfn>
+ <dd>A <a for=/>request</a>
+
+ <dt><dfn export for="deferred fetch record">inactivity deferred delay</dfn> (default null)
+ <dd>Null or a <a>duration</a>
+
+ <dt><dfn export for="deferred fetch record">pending steps</dfn> (default null)
+ <dt><dfn export for="deferred fetch record">invoked callback</dfn> (default null)
+ <dd>Null or an algortihm accepting nothing
+
+ <dt><dfn export for="deferred fetch record">invoked</dfn> (default false)
+ <dd>A boolean
+</dl>
+
+
 <hr>
 
-<p>When a <a for=fetch>fetch group</a> is
-<dfn export for="fetch group" id=concept-fetch-group-terminate>terminated</dfn>, for each associated
-<a for="fetch group">fetch record</a> whose <a for="fetch group">fetch record</a>'s
-<a for="fetch record">controller</a> is non-null, and whose <a for="fetch record">request</a>'s
-<a>done flag</a> is unset or <a for=request>keepalive</a> is false,
-<a for="fetch controller">terminate</a> the <a for="fetch group">fetch record</a>'s
-<a for="fetch record">controller</a>.
+<p>When a <a for=fetch>fetch group</a> <var>fetchGroup</var> is
+<dfn export for="fetch group" id=concept-fetch-group-terminate>terminated</dfn>:
 
+<ol>
+ <li>
+  <p><a for=list>For each</a> <a for="fetch group">deferred fetch record</a>
+  <var>deferredRecord</var> in <var>fetchGroup</var>'s
+  <a for="fetch group">deferred fetch records</a> whose <a for="deferred fetch record">invoked</a> is
+  false:
+
+  <ol>
+   <li><p>If <var>deferredRecord</var>'s <a for="deferred fetch record">pending steps</a> is not
+   null then <a>abort</a> <var>deferredRecord</var>'s
+   <a for="deferred fetch record">pending steps</a>.
+
```

Comment:
There were some privacy requirements from the PendingBeacon API discussion about whether a pending beacon requests should be sent or not depending on user's periodic background sync setting for a site and the status of whether there are still same-site instances open: Details at [this PR](https://github.com/WICG/pending-beacon/pull/55/commits/feb3cf968933cebc823c4c6855d89f07bcf4005a) and [this comment](https://github.com/WICG/pending-beacon/issues/30#issuecomment-1333869614)

Should we include them here in the fetch spec as a condition?

Reply:
I don't think so as periodic sync doesn't currently have cross-vendor support AFAIK.
I reworded to say that the user agent SHOULD fetch the request but can make decisions based on security etc